fuzz/fuzz_targets/general.rs

```diff
@@ -0,0 +1,38 @@
+#![no_main]
+
+use arbitrary::Arbitrary;
+use gfx_fuzz::*;
+use gfx_memory::{GeneralAllocator, GeneralConfig};
+use hal::{memory::Properties, MemoryTypeId};
+
+#[derive(Debug)]
+struct FuzzingInput {
+    config: GeneralConfig,
+    allocations: Vec<Allocation>,
+}
+
+impl Arbitrary for FuzzingInput {
+    fn arbitrary(u: &mut arbitrary::Unstructured) -> Result<Self, arbitrary::Error> {
+        let config = GeneralConfig {
+            block_size_granularity: *u.choose(POWERS_OF_TWO)?,
+            max_chunk_size: *u.choose(POWERS_OF_TWO)?,
+            min_device_allocation: *u.choose(POWERS_OF_TWO)?,
+        };
+        // Need to make sure the values make some sense
+        if config.min_device_allocation > config.max_chunk_size {
+            return Err(arbitrary::Error::IncorrectFormat);
+        }
```

Comment:
I'm not sure about these lines.

If I don't constraint the values to be powers of two, then 99.9% of the time the fuzzer will generate random values, trigger an assertion, and then panic. But I think these are not "interesting" crashes.

Reply:
Only the `block_size_granularity` should be a power of two. Could you try leaving the other ones non-power-of-two?

Reply:
I've tried setting it to something arbitrary and got
```
thread '<unnamed>' panicked at 'Max chunk size must be power of two',
```
This seems to come from an assertion in `GeneralAllocator::new`. Is this not intended?

Reply:
sorry, I missed that one